### PR TITLE
[pointer_interceptor] Removed const keyword from PointerInterceptor constructor.

### PR DESCRIPTION
**Pull request by balvinderz**
Removed const keyword from PointerInterceptor constructor.

https://github.com/flutter/flutter/issues/115652

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [X] I signed the [CLA].
- [X] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [X] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [X] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by balvinderz** [comment]
im-back-woody

**Comment by balvinderz** [comment]
Ya that is much better. I didnt know that mobile implementation was noop

File: packages/pointer_interceptor/lib/src/mobile.dart [hunk 1]
```diff
@@ -2,12 +2,14 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+
```
Comment (ditman) [inline packages/pointer_interceptor/lib/src/mobile.dart:5]:
```suggestion
```

I think this extra line is breaking the dart format check!

**Comment by ditman** [comment]
@balvinderz have you verified that this fix satisfies the original issue? thanks!!

**Comment by balvinderz** [comment]
WhatsApp Image 2023-01-22 at 10 57 12 AM
@ditman  yes warning does not come after making this change.

**Comment by ditman** [comment]
Requested text exemption from @Hixie in discord.

**Comment by ditman** [comment]
@Hixie it's unfortunate, because in this case what we need to do is some once-in-the-lifetime-of-the-library initialization so Flutter web is able to render these widgets. It's literally a static method call in the constructor that only runs once in the initialization of the app.

Other than this, the constructor of the widget could be `const`:

https://github.com/flutter/packages/blob/8f333c455fb9ce854f6cd48d57667fe9b11a1f84/packages/pointer_interceptor/lib/src/web.dart#L44-L48

(`_registered` and `_register` are `static`).

I couldn't find any way of doing that `_register` when the user imports the class (without asking all users to run the `_register()` method manually in their `main` or something).

**Review by ditman — APPROVED** [review]
Let's go!

**Comment by Hixie** [comment]
The usual way of doing it would be in `build` or `initState`.

**Comment by ditman** [comment]
I think I can move the registration to the `build` method, but if I make the constructor `const`, I start getting errors in very normal code, like this:

```dart
  const PointerInterceptor(
    intercepting: false,
    child: ElevatedButton(
      key: Key('wrapped-transparent-button'),
      child: Text('Never calls onPressed transparent'),
      onPressed: () {
                 ^^^^
        _clickedOn('wrapped-transparent-button');
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      },
^^^^^^^
    ),
  ),
```

`onPressed`: Invalid constant value. `dart(invalid_constant)`

I guess it can be made to work, but it seems like a more breaking change than the current solution.

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Pull request merged**